### PR TITLE
feat: Add desktop home page (fix #2665)

### LINKED ISSUE #2665: Empty type query param in search from homepage
### Describe the problem and steps to reproduce it:
When searching from the home page, an empty `type` query param is added to the URL. See: https://github.com/mozilla/addons-frontend/issues/2513#issuecomment-312585463

### What happened?
An empty `type` param is in the URL.

### What did you expect to happen?
No type param is set when it's empty.

### PR DESCRIPTION
**Pull request by tofumatt**
Converts the existing home content to work on desktop... we'll get carousels and such in the future because this was just about porting existing mobile stuff over. But hey–we did it. On the last day. If someone reviews this 😉 😅 

### Mobile
screen shot 2017-06-30 at 11 46 31

### Medium
screen shot 2017-06-30 at 12 34 06

### Large
screen shot 2017-06-30 at 12 34 13

### CODE REVIEW

**Comment by muffinresearch** [comment]
I wonder if the medium layout should the heading and explantion text aligned to the top of the module rather than vertically centered?

**Comment by tofumatt** [comment]
I tried that and it looked kind of awkward:

screenshot 2017-07-01 20 24 34

File: src/amo/components/Home/styles.scss [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  &::before {
+    content: '';
+    display: inline-block;
+    width: 16px;
+    height: 16px;
+    margin: 0 5px;
+  }
+}
+
+.Home-extensions-link:link::before {
```
Comment (muffinresearch) [inline src/amo/components/Home/styles.scss]: Not that there's an obvious alternative afaict.

File: tests/unit/amo/containers/TestHome.js [hunk 2]
```diff
@@ -1,31 +1,104 @@
...
-      'You can change how Firefox works…',
-      '…or what it looks like',
-    ];
-    Array.from(rootNode.querySelectorAll('.HomePage-subheading'))
-      .map((el, index) => expect(el.textContent).toEqual(content[index]));
+  function render(props) {
+    const fakeDispatch = sinon.stub();
+
+    return shallow(
+      <HomeBase
```
Comment (tofumatt) [inline tests/unit/amo/containers/TestHome.js]: Hahaha I thought that too. ([For those not in the UK](http://www.homebase.co.uk/))

File: src/amo/components/Home/styles.scss (hunk 1 above)
Comment (tofumatt) [inline src/amo/components/Home/styles.scss]: Yeah I really forget why we're using `:link` on `a` styles but my goodness is it frustrating.

File: src/amo/components/Home/index.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            {i18n.gettext('Browse all extensions')}
+          </Link>}
+        >
+          <div className="Home-text-wrapper">
+            <h2 className="Home-subheading">
+              {i18n.gettext('You can change how Firefox works…')}
+            </h2>
+            <p className="Home-description">
+              {i18n.gettext(
+                'Install powerful tools that make browsing faster and safer, add-ons make your browser yours.')}
```
Comment (tofumatt) [inline src/amo/components/Home/index.js]: Oh good call, will do.

File: src/amo/components/Home/index.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            {i18n.gettext('Be social')}
+          </ExtensionLink>
+          <ExtensionLink name="play-games" slug="sports-games">
+            {i18n.gettext('Play games')}
+          </ExtensionLink>
+        </ul>
+      );
+    }
+
+    if (clientApp === CLIENT_APP_FIREFOX) {
```
Comment (tofumatt) [inline src/amo/components/Home/index.js]: Yeah, the thing is the category API just returns everything... I guess we could request filters on it though.

**Pull request merged**